### ISSUE TITLE
Best approach for structuring a project?

### ISSUE THREAD

**Issue by PieroSgri** [aws/chalice#1152]
Being new to aws and chalice I'm a bit lost on some points...
I'm developing a project where, in short, there's an arduino uploading stuff on s3 buckets and I have to do multiple things to those files.

Should I use a single chalice app with lot of functions in it using chalice routing? 
Should I use multiple lambdas, separating each lambda in a single chalice app?

On the costs perspective, will a single "big" app be more expensive since it has more code in it and thus will run slower?
What's exactly the difference between chalice routing and having multiple lambdas?

What happen if I have multiple .py files along with the app.py in the chalice project directory when I launch the `chalice deploy` command?

Those questions may sound really naive to experienced developer but there's not that much material online about chalice and aws, most of the material or tutorials refers to broad arguments and is being hard for me to choose an effective approach right at the start to avoid problems later.

**Comment by kmehran1106** [comment]
If you have a single big app, then the time it requires to deploy the full app will increase. Also, there is a restriction on how big your lambda functions can be as well (although I don't think you need to worry about that for your use case). Things should not run slower if you have multiple functions in one app/project.

When you have one app with multiple routes, then all the routes will be created under the same API Gateway (I think you should have already noticed that). As far as I know (and I don't know much) if you have 3 view functions (i.e. APIs) in one project then there will be only one AWS lambda for the three APIs. 
    
    app = Chalice(app_name='my_apis')

    @app.route('/f1', methods=['GET', 'POST'])
    def function1():
        ...
    @app.route('/f2', methods=['PUT', 'DELETE']):
    def funtion2():
        ...    

In this case, you should see one lambda function my_apis (in AWS Management Console -> Lambda). You can see the routes in API Gateway. 

Basically what I've seen, all the functions with the decorator app.route will be kept under the same Lambda for a single project.

Hope this helps!

**Comment by PieroSgri** [comment]
Okay thanks, at last I chose the approach of splitting the project in multiple "chalice app" and to have context related functions in the same app.
For example, a resizer app that automatically resize images uploaded on s3 with all functions related to resizing organized with routing, and so on with other app/lambdas that execute other tasks.
I think is the best approach at least for my use case, a good trade-off I think..

**Comment by stealthycoin** [comment]
@kmehran1106 is correct. All the routes will trigger the same Lambda function, whereas separate `@app.lambda_function()` decorators will be additional lambda functions. None of this will affect the cost of anything as you are only charged for execution time. I would expand on this by adding that putting things in chalicelib won't hurt your execution time or make the project any slower, it just makes it easier to organize and reuse parts of it.

There are a few tutorials located here: https://chalice-workshop.readthedocs.io/en/latest/ that may give you a better idea of how to structure things.
